inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -38,17 +39,222 @@ export class CanvasGraphService {
         // First we clear the canvas so we can draw our data!
         this.clear();
 
-        // TODO: Draw each dataset
-        this.datasets.forEach((dataset: IPerfDataset) => {
-            // do drawing
+        // Get global min max of time axis (across all datasets).
+        let globalTimeMinMax = {min: Infinity, max: 0};
+
+        // TODO: Make better sliding window code (accounting for zoom and pan).
+        // Keep only visible and non empty datasets and get a certain window of items.
+        const datasets = this.datasets.filter((dataset: IPerfDataset) => !dataset.hidden && dataset.data.length > 0).map((dataset: IPerfDataset) => ({
+            ...dataset,
+            data: dataset.data.slice(Math.max(dataset.data.length - 300, 0))

```

Comment:
is the `300` limit fixed? can it be changed / configured by the user in the future?

Reply:
For now** yes, I'm still working out how zooming and panning will work but my thoughts are the user would be manipulating this formula.

I'm not entirely sure how useful being able to configure the initial size of the sliding window may be, but I could add the setting in very easily if wanted

Reply:
awesome. just keep track on it.